crates/ruma-state-res/src/lib.rs

```diff
@@ -267,13 +267,41 @@ where
     F: Fn(&EventId) -> Result<(Int, MilliSecondsSinceUnixEpoch)>,
     Id: Clone + Eq + Ord + Hash + Borrow<EventId>,
 {
-    #[derive(PartialEq, Eq, PartialOrd, Ord)]
+    #[derive(PartialEq, Eq)]
     struct TieBreaker<'a, Id> {
-        inv_power_level: Int,
-        age: MilliSecondsSinceUnixEpoch,
+        power_level: Int,
+        origin_server_ts: MilliSecondsSinceUnixEpoch,
         event_id: &'a Id,
     }
 
+    impl<'a, Id> Ord for TieBreaker<'a, Id>
+    where
+        Id: Ord,
+    {
+        fn cmp(&self, other: &Self) -> Ordering {
+            // NOTE: the power level comparison is "backwards" intentionally.
+            // See the "Mainline ordering" section of the Matrix specification
+            // around where it says the following:
+            //
+            // > for events `x` and `y`, `x < y` if [...]
+            //
+            // <https://spec.matrix.org/v1.12/rooms/v11/#definitions>
+            Ordering::Equal
```

Comment:
This line is unnecessary, we can start with the comparison of power levels.

Reply:
I mean, I didn't write this because I thought it was *necessary*.